### PR TITLE
cmd: add prefix to SYSTEMD_SYSTEM_GENERATOR_DIR

### PR DESCRIPTION
**Pull request by sd-hd**
put systemd generator to /usr/local/lib/systemd/system-generators/*

### CODE REVIEW

File: cmd/configure.ac [hunk 1]
```diff
@@ -214,8 +214,8 @@ AC_ARG_WITH([host-arch-32bit-triplet],
 AC_SUBST(HOST_ARCH32_TRIPLET)
 AC_DEFINE_UNQUOTED([HOST_ARCH32_TRIPLET], "${HOST_ARCH32_TRIPLET}", [Arch triplet for 32bit libraries])
 
-SYSTEMD_SYSTEM_GENERATOR_DIR="$($PKG_CONFIG --variable=systemdsystemgeneratordir systemd)"
-AS_IF([test "x$SYSTEMD_SYSTEM_GENERATOR_DIR" = "x"], [SYSTEMD_SYSTEM_GENERATOR_DIR=/lib/systemd/system-generators])
+SYSTEMD_SYSTEM_GENERATOR_DIR="${prefix}$($PKG_CONFIG --variable=systemdsystemgeneratordir systemd)"
```
Comment (zyga) [inline cmd/configure.ac]: I'm not super convinced this is something that needs to have a prefix. This is not our directory (hence not a local-snapd-installation type decision) but a systemd directory. Does systemd pick up generators from local directories?

Comment (sd-hd) [reply cmd/configure.ac]: I found/usr/local/lib/systemd/system-generators/* in this documentation:

https://www.freedesktop.org/software/systemd/man/systemd.generator.html

And, if i run "sudo make uninstall": then my system generator was deleted :-(

Comment (zyga) [reply cmd/configure.ac]: After discussing with Maciek I think he's right to point out that the prefix of snapd build and the system-wide install systemd directory path are unrelated.

If you want to get the "local" location it needs to be conveyed some other way, perhaps with a special case or in downstream packaging.

Comment (bboozzoo) [reply cmd/configure.ac]: I think the change is wrong. What happens is that the code runs pkg-config to determine which location the systemd package in the system expects to find the unit files and returns with an answer. There is no indication whether `systemd` will look for units files under `$prefix/$systemdsystemgeneratordir` (prefix can be a totally random thing) and AFAIK there's no way to ask whether such path is looked into by systemd.

If you just want to install all the files under some tree, pass `DESTDIR`, eg. `make DESTDIR=$PWD/install-dir install`.

**Comment by sd-hd** [comment]
Hi Zyga, 

for me it's ok to close this.

But then, we should also remove the prefix from the lines after that:

```
# FIXME: get this via something like pkgconf once it is defined there
SYSTEMD_SYSTEM_ENV_GENERATOR_DIR="${prefix}/lib/systemd/system-environment-generators"
AC_SUBST(SYSTEMD_SYSTEM_ENV_GENERATOR_DIR)

```

Should i change this ?

**Comment by bboozzoo** [comment]
This one is an exception. There is dedicated variable in `systemd.pc` that defines where the system env generators are to be placed. We should probably fix it to something like this:
```
SYSTEMD_SYSTEM_ENV_GENERATOR_DIR="$($PKG_CONFIG --variable=systemdutildir systemd)/system-environment-generators"
```

And it'd be nice to add a proper variable in for env generators in systemd upstream as well.

**Comment by sd-hd** [comment]
> And it'd be nice to add a proper variable in for env generators in systemd upstream as well.

like @bboozzoo says, it should be first fixed upstream

**Comment by pedronis** [comment]
as per discussion, if anything this needs to be addressed upstream in systemd, closing

**Comment by bboozzoo** [comment]
@sd-hd I suggest reading out systemd upstream and opening a issue with a question as to which variable the consumers should look at. Maybe suggest to add a new variable for the generators. Bonus points for landing a PR upstream :smile:

**Comment by sd-hd** [comment]
@bboozzoo
I want try. Already had forked the systemd repo ;)

**Comment by sd-hd** [comment]
@bboozzoo 
i hope, i could help  
https://github.com/systemd/systemd/pull/14600

**Pull request closed without merging**